File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.cpp
#include <string>
#include <iostream>
using namespace std;

bool solution(string s) {
    
    for (int i = 0; i < s.length(); i++) {
        s[i] = tolower(s[i]);
    }

    int count = 0;

   
    for (int i = 0; i < s.length(); i++) {
        if (s[i] == 'p') {
            count++;
        } else if (s[i] == 'y') {
            count--;
        }
    }

    return count == 0;
}

File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.java
class Solution {
boolean solution(String s) {
		boolean answer = true;
		
		String word = s.toLowerCase();
		
		int count = 0;
		
		for (int i = 0; i < word.length(); i++) {
            if (word.charAt(i) == 'p') {
                count += 1;
            } else if (word.charAt(i) == 'y') {
                count -= 1;
            }
        }
		
		if(count == 0) {
			answer = true;
		}else {
			answer = false;
		}
		
		return answer;
	}
}